sched/irq/irq_csection.c

```diff
@@ -75,6 +75,18 @@ volatile uint8_t g_cpu_nestcount[CONFIG_SMP_NCPUS];
  * Private Functions
  ****************************************************************************/
 
+static bool csection_degraded(void)
```

Comment:
Additional checks will degrade SMP performance, why not add such checks to the syslog path?

cc @hujun260 ,
os ready checks removed by PR:
https://github.com/apache/nuttx/pull/13485

Reply:
@yamt 
The invocation of critical sections is still quite frequent. Previously, we removed those checks in order to enhance performance.
In some specific scenarios, we can use spin_lock_irqsave to replace the critical sections or modify the business logic judgments.
So far, I haven't encountered situations where such modifications are absolutely necessary.

Reply:
https://github.com/apache/nuttx/pull/12281 this patch  remove the  judgement

Reply:
i added the check to enter_critical_section because i felt it logically belonged there, not the caller.

> https://github.com/apache/nuttx/pull/12281 this patch remove the judgement

i was not aware of the history. if we decided that enter_critical_section should not be used early in the boot, it's better to have a DIAGASSERT there to assert it. how do you think?

Reply:
We have a long-term plan to reduce the number of calls and the scope of critical sections, as well as to decrease critical sections complexity.
A lot of work has already been done in this area, so we do not wish to add any logic within the critical sections.

Reply:
any logic, including DIAGASSERT, you mean? why?

Reply:
> any logic, including DIAGASSERT, you mean? why?

This line indicate that enter_critical_section should not be used early in the boot
https://github.com/apache/nuttx/blob/d4f6cc229d55a0c252ba6a821b9f1789e53d4bbc/sched/irq/irq_csection.c#L206

Reply:
> Additional checks will degrade SMP performance, why not add such checks to the syslog path?

i moved it to syslog because it seems controversial to modify critical section logic.